graphql/resolvers/PublicUser.js

```diff
@@ -1,11 +1,70 @@
 module.exports = {
-  async testimonial (user, args, {pgdb}) {
-    const testimonial = await pgdb.public.testimonials.findOne({userId: user.id})
+  async testimonial (user, args, { pgdb }) {
+    const testimonial = await pgdb.public.testimonials.findOne({
+      userId: user.id
+    })
     if (testimonial) {
       return {
         ...testimonial,
         name: user.name
       }
     }
+  },
+  async latestComments (user, args, { pgdb }) {
+    const userId = user.id
+    const limit = 10
+
+    const comments = await pgdb.query(
+      `
+      SELECT
+        c.id,
+        c."userId",
+        c.content,
+        c."adminUnpublished",
+        c.published,
+        c."createdAt",
+        c."updatedAt",
+        c."discussionId",
+        d.title AS "discussionTitle"
+      FROM comments c
+      JOIN discussions d ON d.id = c."discussionId"
+      WHERE
+        c."userId" = :userId
+      ORDER BY
+        c."createdAt" DESC
+      LIMIT :limit;
+    `,
+      { userId, limit }
+    )
+
+    if (comments.length) {
+      // Respect anonymous settings, if any.
```

Comment:
have a look at graphql/resolvers/Comment.js
It resolves the displayUser while respecting preferences and settings of the commenter and the discussion. It think you can just return the comments and things will be fine.